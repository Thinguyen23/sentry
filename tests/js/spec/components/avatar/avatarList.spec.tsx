import React from 'react';
import {render} from '@testing-library/react';

import AvatarList from 'app/components/avatar/avatarList';
import {AvatarUser} from 'app/types';

const avatarUsers: Array<AvatarUser> = [
  {
    id: '1',
    name: 'Jan',
    username: 'jan_sentry',
    email: 'jan_sentry@gmail.com',
    ip_address: '12.31.20.11',
  },
  {
    id: '2',
    name: 'Daniel',
    username: 'daniel_sentry',
    email: 'daniel_sentry@gmail.com',
    ip_address: '12.31.20.12',
  },
];

function renderComponent(avatarUsersSixUsers: AvatarList['props']['users']) {
  const utils = render(<AvatarList users={avatarUsersSixUsers} />);
  return {...utils};
}

describe('AvatarList', () => {
  it('renders with user letter avatars', () => {
    const {container, queryByTestId, getByText} = renderComponent(avatarUsers);
    expect(getByText('D')).toBeTruthy();
    expect(getByText('J')).toBeTruthy();
    expect(queryByTestId('avatarList-collapsedusers')).toBeNull();
    // too get the root element of your rendered element, use container.firstChild
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with collapsed avatar count if > 5 users', () => {
    const avatarUsersSixUsers = [
      ...avatarUsers,
      {
        id: '3',
        name: 'Matej',
        username: 'matej_sentry',
        email: 'matej_sentry@gmail.com',
        ip_address: '12.31.20.13',
      },
      {
        id: '4',
        name: 'Billy',
        username: 'billy_sentry',
        email: 'jbilly_sentry@gmail.com',
        ip_address: '12.31.20.14',
      },
      {
        id: '5',
        name: 'Mark',
        username: 'mark_sentry',
        email: 'mark_sentry@gmail.com',
        ip_address: '12.31.20.15',
      },
      {
        id: '6',
        name: 'Alberto',
        username: 'alberto_sentry',
        email: 'alberto_sentry@gmail.com',
        ip_address: '12.31.20.16',
      },
    ];

    const {container, getByTestId, queryByText, queryAllByText} = renderComponent(
      avatarUsersSixUsers
    );
    expect(queryAllByText(avatarUsersSixUsers[4].name.charAt(0))).toBeTruthy();
    expect(queryByText(avatarUsersSixUsers[3].name.charAt(0))).toBeTruthy();
    expect(queryByText(avatarUsersSixUsers[5].name.charAt(0))).toBeNull();
    expect(getByTestId('avatarList-collapsedusers')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
