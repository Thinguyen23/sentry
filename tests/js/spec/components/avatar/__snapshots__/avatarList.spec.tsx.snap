// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarList renders with collapsed avatar count if > 5 users 1`] = `
<div
  class="css-1kbnx78-AvatarListWrapper e1mtf41h0"
>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <div
      class="css-1xikpge-CollapsedUsers-Circle e1mtf41h2"
      data-testid="avatarList-collapsedusers"
      size="28"
    >
      <span
        class="css-jn99p5-Plus e1mtf41h3"
      >
        +
      </span>
      1
    </div>
  </span>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Jan"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#6c5fc7"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          J
        </text>
      </svg>
    </span>
  </span>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Daniel"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#f9a66d"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          D
        </text>
      </svg>
    </span>
  </span>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Matej"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#6c5fc7"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          M
        </text>
      </svg>
    </span>
  </span>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Billy"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#ec5e44"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          B
        </text>
      </svg>
    </span>
  </span>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Mark"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#57be8c"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          M
        </text>
      </svg>
    </span>
  </span>
</div>
`;

exports[`AvatarList renders with user letter avatars 1`] = `
<div
  class="css-1kbnx78-AvatarListWrapper e1mtf41h0"
>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Jan"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#6c5fc7"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          J
        </text>
      </svg>
    </span>
  </span>
  <span
    aria-describedby="tooltip-123456"
    class="css-1y4rfpo-Container eowlwvy0"
  >
    <span
      class="avatar e1mtf41h1 css-11cenzn-StyledBaseAvatar-StyledAvatar-Circle e147uwb0"
      style="width: 28px; height: 28px;"
      title="Daniel"
      type="letter_avatar"
    >
      <svg
        class="css-1oxr9oh-LetterAvatar e12lgfev0"
        viewBox="0 0 120 120"
      >
        <rect
          fill="#f9a66d"
          height="120"
          rx="15"
          ry="15"
          width="120"
          x="0"
          y="0"
        />
        <text
          fill="#FFFFFF"
          font-size="65"
          style="dominant-baseline: central;"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          D
        </text>
      </svg>
    </span>
  </span>
</div>
`;
